plugins/catalog-backend-module-openapi-spec/src/InternalOpenApiDocumentationProvider.ts

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ApiEntity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { ForwardedError } from '@backstage/errors';
+import {
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+import { merge, isErrorResult } from 'openapi-merge';
+import { getOpenApiSpecRoute } from '@backstage/backend-openapi-utils';
+import type {
+  OpenAPIObject,
+  OperationObject,
+  PathItemObject,
+} from 'openapi3-ts';
+import fetch from 'cross-fetch';
+import { DiscoveryService, LoggerService } from '@backstage/backend-plugin-api';
+import * as uuid from 'uuid';
+import { PluginTaskScheduler, TaskRunner } from '@backstage/backend-tasks';
+
+const HTTP_VERBS: (keyof PathItemObject)[] = [
+  'get',
+  'post',
+  'put',
+  'delete',
+  'patch',
+  'trace',
+  'options',
+  'head',
+];
+
+const addTagsToSpec = (spec: OpenAPIObject, tag: string) => {
+  Object.values(spec?.paths).forEach((path: PathItemObject) => {
+    HTTP_VERBS.forEach(verb => {
+      if (verb in path) {
+        if (!('tags' in path[verb])) {
+          (path[verb] as OperationObject).tags = [];
+        }
+        if (!(path[verb] as OperationObject).tags?.includes(tag)) {
+          (path[verb] as OperationObject).tags?.push(tag);
+        }
+      }
+    });
+  });
+};
+
+const mergeSpecs = async ({
+  baseUrl,
+  specs,
+}: {
+  baseUrl: string;
+  specs: OpenAPIObject[];
+}) => {
+  const mergeResult = merge([
```

Comment:
How much configurability do we want here?

Reply:
Think this might be fine for the start - can be iterated depending on feedback.